plugins/elasticsearch-bboss/src/main/java/com/navercorp/pinpoint/plugin/elasticsearchbboss/ElasticsearchPlugin.java

```diff
@@ -0,0 +1,202 @@
+	public static final AnnotationKey ARGS_RESPONSEHANDLE_ANNOTATION_KEY = AnnotationKeyFactory.of(975, "es.responseHandle", AnnotationKeyProperty.VIEW_IN_RECORD_SET);
+	public static final AnnotationKey ARGS_VERSION_ANNOTATION_KEY = AnnotationKeyFactory.of(976, "es.version", AnnotationKeyProperty.VIEW_IN_RECORD_SET);
+	public static final ServiceType ELASTICSEARCH_EXECUTOR = ServiceTypeFactory.of(9977, "ElasticsearchBBossExecutor");
+	private static final String ELASTICSEARCH_SCOPE = "ElasticsearchBBoss_SCOPE";
+	private static final String ELASTICSEARCH_EXECUTOR_SCOPE = "ElasticsearchBBossExecutor_SCOPE";
+	private static final List<ElasticsearchInterceptorClassInfo> clazzInterceptors = new ArrayList<ElasticsearchInterceptorClassInfo>();
+
+	static {
+		init();
+	}
+	static void init(){
+		ElasticsearchInterceptorClassInfo interceptorClassInfo = null;
+		List<ElasticsearchMethodInfo> methodInfos = null;
+		ElasticsearchMethodInfo methodInfo = null;
+		interceptorClassInfo = new ElasticsearchInterceptorClassInfo();
+		interceptorClassInfo.setInterceptorClass("org.frameworkset.elasticsearch.client.ConfigRestClientUtil");
+		methodInfo = new ElasticsearchMethodInfo();
+		methodInfo.setFilterType(1);
+		methodInfo.setName("*");
+		methodInfo.setPattern(true);
+		interceptorClassInfo.setAllAccept(methodInfo);
+
+		methodInfos = null;
+		methodInfos = new ArrayList<ElasticsearchMethodInfo>();
+		methodInfo = new ElasticsearchMethodInfo();
+		methodInfo.setFilterType(1);
+		methodInfo.setName("*");
+		methodInfo.setPattern(true);
+		methodInfos.add(methodInfo);
+		methodInfo = new ElasticsearchMethodInfo();
+		methodInfo.setFilterType(0);
+		methodInfo.setName("discover");
+		methodInfo.setPattern(false);
+		methodInfos.add(methodInfo);
+		interceptorClassInfo.setInterceptorMehtods(methodInfos);
+		interceptorClassInfo.setMethodFilter(new ElasticsearchCustomMethodFilter(null, interceptorClassInfo));
+		clazzInterceptors.add(interceptorClassInfo);
+		interceptorClassInfo = new ElasticsearchInterceptorClassInfo();
+		interceptorClassInfo.setInterceptorClass("org.frameworkset.elasticsearch.client.RestClientUtil");
+		methodInfo = new ElasticsearchMethodInfo();
+		methodInfo.setFilterType(1);
+		methodInfo.setName("*");
+		methodInfo.setPattern(true);
+		interceptorClassInfo.setAllAccept(methodInfo);
+
+		methodInfos = null;
+		methodInfos = new ArrayList<ElasticsearchMethodInfo>();
+		methodInfo = new ElasticsearchMethodInfo();
+		methodInfo.setFilterType(1);
+		methodInfo.setName("*");
+		methodInfo.setPattern(true);
+		methodInfos.add(methodInfo);
+		methodInfo = new ElasticsearchMethodInfo();
+		methodInfo.setFilterType(0);
+		methodInfo.setName("discover");
+		methodInfo.setPattern(false);
+		methodInfos.add(methodInfo);
+		interceptorClassInfo.setInterceptorMehtods(methodInfos);
+		interceptorClassInfo.setMethodFilter(new ElasticsearchCustomMethodFilter(null, interceptorClassInfo));
+		clazzInterceptors.add(interceptorClassInfo);
```

Comment:
Let's say _interceptorClassInfo_ is Class A and _ElasticsearchCustomMethodFilter_ is B.
Inside A there is Class B. But Inside B there is some part of A in it.
Is this really necessary?
Do you think there can be a better way to resolve this complexity?

Reply:
Thank for your suggests. Elasticsearchbboss plugin is writed based on pinpoint-plugin-generate tool,I'll refacted it soon as your suggest.